src/com/xilinx/rapidwright/design/drc/NetRoutesThruLutAtMostOnce.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021 Xilinx, Inc.
+ * All rights reserved.
+ *
+ * Author: Eddie Hung, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.design.drc;
+
+import com.xilinx.rapidwright.design.Cell;
+import com.xilinx.rapidwright.design.Design;
+import com.xilinx.rapidwright.design.DesignTools;
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SitePinInst;
+import com.xilinx.rapidwright.device.BELPin;
+import com.xilinx.rapidwright.device.Node;
+import com.xilinx.rapidwright.device.PIP;
+import com.xilinx.rapidwright.util.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Check that each LUT contains at most one routethru of each net.
+ * Identifies occurrences of issue #226.
+ * Failed checks will print a warning and are not counted unless the strict parameter is true.
+ */
+public class NetRoutesThruLutAtMostOnce {
+
+    private static String lutName(Node node) {
+        String nodeWireName = node.getWireName();
+        return node.getTile().getName() + "/" + nodeWireName.substring(0, nodeWireName.length() - 1);
+    }
+
+    public static int run(Design design, boolean strict) {
+        List<Pair<Net, List<Pair<String, Integer>>>> netToLutRoutethrus = design.getNets().stream()
+                .map((n) -> {
+                    Map<String, Integer> lutRoutethrus = new HashMap<>();
+
+                    for (PIP p : n.getPIPs()) {
+                        if (!p.isRouteThru()) continue;
+                        // Set to 1 if not exist, increment by 1 if does exist
+                        lutRoutethrus.merge(lutName(p.getStartNode()), 1, Integer::sum);
+                    }
+
+                    for (SitePinInst spi : n.getSinkPins()) {
+                        for (BELPin bp : spi.getSiteWireBELPins()) {
+                            Cell c = spi.getSiteInst().getCell(bp.getBEL());
+                            // Filter out all non routethru cells
+                            if (c == null || !c.isRoutethru()) continue;
+
+                            assert(DesignTools.isBELALut(bp.getBELName()));
```

Comment:
Since asserts are not turned on by default, will this catch the same issues if the assertion is not checked?

Reply:
My thinking here was to (a) subtly indicate to the reader that I am only expecting/assuming a LUT here and (b) enforce that with a string-based check only if assertions were enabled, but not pay for that if assertions were disabled.

Would it be an extra-exceptional (extra'ceptional?) situation if this assumption was to break down?

Reply:
What would happen if the routethru cell was not a LUT?  I think this case is common.  Perhaps I add a `BEL.isLUT()` method as part of #277?

Reply:
I actually discovered that route-thru PIPs exist on IOLOGICs as well, which were being incorrectly considered before and is now fixed.

I think a fast BEL.isLUT() would be very helpful!